js/src/ui/Form/AddressSelect/addressSelect.js

```diff
@@ -15,262 +15,799 @@
+      {
+        label: 'accounts',
+        values: [].concat(
+          Object.values(wallets),
+          Object.values(accounts)
+        )
+      },
+      {
+        label: 'contacts',
+        values: Object.values(contacts)
+      },
+      {
+        label: 'contracts',
+        values: Object.values(contracts)
+      }
+    ];
+
+    this.handleChange();
   }
 
   render () {
-    const { allowInput, disabled, error, hint, label } = this.props;
-    const { autocompleteEntries, value } = this.state;
+    const input = this.renderInput();
+    const content = this.renderContent();
+
+    const classes = [ styles.main ];
+
+    return (
+      <div
+        className={ classes.join(' ') }
+        onBlur={ this.handleMainBlur }
+        onClick={ this.handleFocus }
+        onFocus={ this.handleMainFocus }
+        onKeyDown={ this.handleInputAddresKeydown }
+        ref='inputAddress'
+        tabIndex={ 0 }
+      >
+        { input }
+        { content }
+      </div>
+    );
+  }
 
-    const searchText = this.getSearchText();
-    const icon = this.renderIdentityIcon(value);
+  renderInput () {
+    const { focused } = this.state;
+    const { accountsInfo, disabled, error, hint, label, value } = this.props;
+
+    const input = (
+      <InputAddress
+        accountsInfo={ accountsInfo }
+        error={ error }
+        focused={ focused }
+        label={ label }
+        hint={ hint }
+        tabIndex={ -1 }
+        value={ value }
+
+        allowCopy={ false }
+        disabled
+        text
+      />
+    );
+
+    if (disabled) {
+      return input;
+    }
 
     return (
-      <div className={ styles.container }>
-        <AutoComplete
-          className={ !icon ? '' : styles.paddedInput }
-          disabled={ disabled }
-          label={ label }
-          hint={ hint ? `search for ${hint}` : 'search for an address' }
-          error={ error }
-          onChange={ this.onChange }
-          onBlur={ this.onBlur }
-          onUpdateInput={ allowInput && this.onUpdateInput }
-          value={ searchText }
-          filter={ this.handleFilter }
-          entries={ autocompleteEntries }
-          entry={ this.getEntry() || {} }
-          renderItem={ this.renderItem }
-        />
-        { icon }
+      <div
+        className={ styles.inputAddress }
+      >
+        { input }
       </div>
     );
   }
 
-  renderIdentityIcon (inputValue) {
-    const { error, value, label } = this.props;
+  renderContent () {
+    const { muiTheme } = this.context;
+    const { hint, disabled, label } = this.props;
+    const { expanded, top, left, inputFocused } = this.state;
 
-    if (error || !inputValue || value.length !== 42) {
+    if (disabled) {
       return null;
     }
 
-    const classes = [ styles.icon ];
+    const classes = [ styles.overlay ];
 
-    if (!label) {
-      classes.push(styles.noLabel);
+    if (expanded) {
+      classes.push(styles.expanded);
     }
 
+    const id = `${label}_${hint}`;
```

Comment:
Will this work with `FormattedMessages`? Looks like we will end up with an `[Object object]_[Object object]` situation here